apispec/ext/marshmallow/swagger.py

```diff
@@ -41,6 +41,32 @@
     marshmallow.fields.List: ('array', None),
 }
 
+CUSTOM_FIELD_MAPPING_ATTR = '__custom_swagger_field_mapping'
+
+
+def map_to_swagger_field(*args):
+    """
+    decorator to set mapping for custom fields.
+
+    args can be:
+
+    - a core marshmallow field type (in which case we reuse that type's mapping)
+    - a pair of string|None
+    """
+    if len(args) == 1 and args[0] in FIELD_MAPPING:
+        swagger_type_field = FIELD_MAPPING[args[0]]
+    elif len(args) == 2:
+        swagger_type_field = args
+    else:
+        raise ArgumentError('pass core marshmallow field type or (type, fmt) pair')
```

Comment:
I think a `TypeError` is more appropriate here. Also, uppercase the first letter of the message.